fix and re-land "re-enable `android_view_test` (#54214)"

cyanglaz opened the pull request:
## Description

The tests were broken because each test assume it starts from the home page but none return to the home page after finishing.
Added mechanisms for each test page to go back to the home page.

## Related Issues

https://github.com/flutter/flutter/issues/54022

## Tests

I added the following tests:
`'MotionEvent recomposition`

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

Commit:
fix android view test

amirh commented on dev/integration_tests/android_views/android/app/src/main/java/io/flutter/integration/androidviews/SimplePlatformView.java:
```diff
@@ -52,21 +52,27 @@ public void onMethodCall(MethodCall methodCall, MethodChannel.Result result) {
         AlertDialog.Builder builder = new AlertDialog.Builder(context);
         TextView textView = new TextView(context);
-        textView.setText("Alert!");
+        textView.setText("This alert dialog will close in 1 seconds");
```
nit: 1 second (probably my original typo)

amirh commented on dev/integration_tests/android_views/lib/motion_events_page.dart:
```diff
@@ -63,52 +63,60 @@ class MotionEventsBodyState extends State<MotionEventsBody> {
-          child: ListView.builder(
-            itemBuilder: buildEventTile,
-            itemCount: flutterViewEvents.length,
+        children: <Widget>[
```
can you keep the original indentation? it's hard to see what has changed

Commit:
Update SimplePlatformView.java

Commit:
fix format

Commit:
:wq!
Merge branch 'android_view_test_fix' of github.com:cyanglaz/flutter into android_view_test_fix

Commit:
remove unnecessary spacing params

Commit:
revert

cyanglaz commented:
@amirh done!

amirh reviewed (approved):
LGTM
